### PR TITLE
Patch the .AppImage with known working libs to fix openssl crash

### PR DESCRIPTION
**Pull request by meriadec**
This commit fixes the crash occuring on some configuration (reported only on Fedora 28, but since it comes from a library conflict, it can affect a larger spectrum) when syncing BTC-like account.

To give more context, libcore is using libcrypto symbols, and should use the libcore-embedded libcrypto, *BUT* electron is loading the system one before and it result in a SIGSEGV when using the Fedora 28 openssl version ([list here](https://rpmfind.net/linux/rpm2html/search.php?query=libcrypto.so.1.1()(64bit))). More details on the crash can be found on the original issue.

With @KhalilBellakrid, we first tried to address the problem by changing the runtime path of lib-ledger-core (changing `LD_LIBRARY_PATH` when launching the app was solving the issue.. but it could not be made persistent). Even with successfully changing the `RPATH` inside binaries, the incorrect `libcrypto` was still loaded, and that's because.. electron load it before everything.

**TL;DR:** The fix is hacky. It basically unpack the `.AppImage`, put required libs inside, then repack it. Not perfect. But the issue has been here for too long, and has led to lot of deception. I've tested it against auto-update and I confirm it works. (As you see in the commits, I couldn't achieve to interject _inside_ `electron-builder` workflow, so had to add dedicated script to upload gh assets.. hope we find a way to get rid of that).

*(poor-quality) video of the fix in action:*

[](https://www.youtube.com/watch?v=rl5GXXdjkUY)

### Type

Bugfix

### Context

https://github.com/LedgerHQ/ledger-live-desktop/issues/1010

### Parts of the app affected / Test plan

Parts of the app affected

To reproduce:
- install Fedora 27
- upgrade to Fedora 28 (via `dnf upgrade`, etc.) - *note to @Arnaud97234 : we have a machine on this configuration)*
- install any release before this fix (e.g [v1.2.0](https://github.com/LedgerHQ/ledger-live-desktop/releases/tag/v1.2.0))
- sync BTC-like account
- **should fail** with `Internal process error (null)` (openssl crash, called by libcore)
- build the app with `yarn dist`
- patch the app with `bash scripts/patch-appimage.sh`
- launch the app
- **should work**

### CODE REVIEW

**Comment by Arnaud97234** [comment]
Tests done.
BTC like coin are working properly : Sync, Operations.
No regressions on other coins (ETH, XRP ...)

**Comment by meriadec** [comment]
As AppImageKit use [continuous](https://github.com/AppImage/AppImageKit/releases) release system, it would be problematic to store the shasum because it could cause false positive if they push a new release. One solution will be to store our own version on S3 :D WDYT ?

**Comment by KhalilBellakrid** [comment]
Yes I thought also about that and I think the best solution would be to push those libraries from Libcore CI to s3 bucket (with their checksum) so that way we always have 'compliant' versions and we get the updates if there are some.

**Comment by meriadec** [comment]
appimagekit will be pushed to S3 tomorrow by infra team.

@gre do we wait or do we merge for now?

**Comment by gre** [comment]
let's wait it's ready
